src/test/java/com/amazon/opendistroforelasticsearch/search/asynchronous/integTests/AsynchronousSearchRejectionIT.java

```
@@ -122,7 +126,11 @@ public void onFailure(Exception e) {
                                                             numRejections.incrementAndGet();
                                                         } else if (cause instanceof ElasticsearchTimeoutException) {
                                                             numTimeouts.incrementAndGet();
+                                                        } else if(e instanceof ResourceNotFoundException) {
+                                                            // deletion is in race with task cancellation due to partial merge failure
+                                                            numRnf.getAndIncrement();
                                                         } else {
+                                                            logger.error("Unexpected failure : ", e);
```

fail instead of log?